### PR TITLE
Task-8: Pagination

### PR DESCRIPTION
**Pull request by mza-wtag**
[Task 8: Pagination](https://trello.com/c/gp8gErvt/12-task-8-pagination)

**Small Description:** As a user I want see 9 blogs initially. Then clicking on see more button will show more.

## Types of changes
- [x] Initially show 9 blogs.
- [x] If total blogs are more than 9 than show a show more button.
- [x] Clicking on Show More button will show 6 more blogs each time.

## Review
Please have a close look at this PR and kindly put your valuable comments.

## Screen Shot
image
image

## Screen recording
[Screencast from 02-15-2024 01:36:03 PM.webm](https://github.com/mza-wtag/my-blog-react-app/assets/151821213/d161cae3-7f1d-4ea5-9312-68dc86faa4b0)

### CODE REVIEW

**Review by nmi-wtag — APPROVED** [review]

File: src/components/BlogList/BlogList.jsx [hunk 1]
```diff
@@ -1,27 +1,43 @@
...
 import { useSelector } from "react-redux";
 import BlogCard from "@components/BlogCard/BlogCard";
 import NotFound from "@components/NotFound/NotFound";
+import Button from "@components/Button/Button";
 import "@components/BlogList/blogList.scss";
 
 const BlogList = ({ blogs }) => {
   const searchQuery = useSelector((state) => state.search.searchQuery);
   const filteredTag = useSelector((state) => state.filter.filteredTag);
+  const [blogsToShow, setBlogsToShow] = useState(9);
```
Comment (maa-wtag) [inline src/components/BlogList/BlogList.jsx]: put the default Blogs to Show count to a variable like `DEFAULT_BLOGS_TO_SHOW`

File: src/components/BlogList/BlogList.jsx [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
 
   const filteredBlogs = blogs?.filter((blog) => {
     const titleMatches = blog.title.includes(searchQuery.toLowerCase());
     const tagMatches = !filteredTag || blog.tags.includes(filteredTag);
     return titleMatches && tagMatches;
   });
 
+  const handleShowMore = () => {
+    setBlogsToShow((prevCount) => prevCount + 6);
```
Comment (maa-wtag) [inline src/components/BlogList/BlogList.jsx]: also put it into another variable like `MORE_BLOGS_TO_SHOW`

File: src/components/BlogList/BlogList.jsx [hunk 3]
```diff
@@ -1,27 +1,43 @@
...
+
   return (
-    <div className="blog-list">
-      {filteredBlogs.length > 0 ? (
-        filteredBlogs.map((blog) => <BlogCard key={blog.id} blog={blog} />)
-      ) : (
-        <NotFound text="No Blogs Found." />
+    <>
+      <div className="blog-list">
+        {filteredBlogs.length > 0 ? (
```
Comment (maa-wtag) [inline src/components/BlogList/BlogList.jsx]:
```suggestion
        {filteredBlogs?.length > 0 ? (
```

File: src/components/BlogList/BlogList.jsx [hunk 4]
```diff
@@ -1,27 +1,43 @@
...
+          filteredBlogs
+            .slice(0, blogsToShow)
+            .map((blog) => <BlogCard key={blog.id} blog={blog} />)
+        ) : (
+          <NotFound text="No Blogs Found." />
+        )}
+      </div>
+      {filteredBlogs.length > blogsToShow && (
```
Comment (maa-wtag) [inline src/components/BlogList/BlogList.jsx]:
```suggestion
      {filteredBlogs?.length > blogsToShow && (
```

File: src/constants/tags.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [
-  { "value": "Technology", "label": "Technology" },
-  { "value": "Poetry", "label": "Poetry" },
-  { "value": "Films", "label": "Films" },
-  { "value": "World Politics", "label": "World Politics" }
+  { "value": " Technology ", "label": "Technology" },
+  { "value": " Poetry ", "label": "Poetry" },
+  { "value": " Films ", "label": "Films" },
+  { "value": " World Politics ", "label": "World Politics" }
```
Comment (maa-wtag) [inline src/constants/tags.json]: why space?

**Review by maa-wtag — CHANGES_REQUESTED** [review]

File: src/constants/tags.json (hunk 5 above)
Comment (mza-wtag) [reply src/constants/tags.json]: If I select multiple tags, then it shows one after another, like one word. I tried to handle it with scss to give a space, but there is no scss element found for this as it came from the json file dynamically.

File: src/components/BlogList/BlogList.jsx (hunk 4 above)
Comment (mza-wtag) [reply src/components/BlogList/BlogList.jsx]: done

File: src/components/BlogList/BlogList.jsx (hunk 3 above)
Comment (mza-wtag) [reply src/components/BlogList/BlogList.jsx]: done

File: src/components/BlogList/BlogList.jsx (hunk 2 above)
Comment (mza-wtag) [reply src/components/BlogList/BlogList.jsx]: done

File: src/components/BlogList/BlogList.jsx (hunk 1 above)
Comment (mza-wtag) [reply src/components/BlogList/BlogList.jsx]: done

**Review by maa-wtag — APPROVED** [review]

**Review by hhu-wtag — APPROVED** [review]

**Pull request merged**